paddle/fluid/operators/activation_op.cc

```diff
@@ -128,6 +128,26 @@ class ActivationOp : public framework::OperatorWithKernel {
       const framework::ExecutionContext& ctx) const override {
     return GetKernelType(ctx, *this, "X");
   }
+
+  framework::OpKernelType GetKernelTypeForVar(
+      const std::string& var_name, const Tensor& tensor,
+      const framework::OpKernelType& expected_kernel_type) const {
+#ifdef PADDLE_WITH_MKLDNN
+    // When activation is first oneDNN op (there was some non oneDNN op
```

Comment:
Do all operators need to adapt to the format of the input NHWC when it is the first mkldnn operator?

Reply:
Yes. So this PR is only doing this change for fixing #38216 , but in general similar change has to be added to other oneDNN ops.